plugins/auto-sizes/hooks.php

```diff
@@ -107,9 +126,18 @@ function auto_sizes_improve_image_sizes_attribute( string $content, array $parse
 
 	// Only update the markup if an image is found.
 	if ( $has_image ) {
-		$layout = wp_get_global_settings( array( 'layout' ) );
-		$align  = $parsed_block['attrs']['align'] ?? null;
-		$sizes  = null;
+		$layout     = wp_get_global_settings( array( 'layout' ) );
+		$align      = $parsed_block['attrs']['align'] ?? null;
+		$image_id   = $parsed_block['attrs']['id'] ?? '';
+		$image_size = $parsed_block['attrs']['sizeSlug'] ?? '';
+
+		$image_attributes = wp_get_attachment_image_src( $image_id, $image_size );
```

Comment:
Instead of calling `wp_get_attachment_image_src()`, we should be able to parse the width attribute directly from the image markup using the existing `WP_HTML_Tag_Processor` instance. 

We could also factor in the `width` attribute from the parsed block, to cover cases where the image has been resized manually in the editor, which is another natural layout constraint.

Reply:
@joemcgill, the editor does not add a width attribute in the image markup, so we can't retrieve it through `WP_HTML_Tag_Processor`.

Check the markup:
```
<!-- wp:image {"id":20,"sizeSlug":"large","linkDestination":"none"} -->
<figure class="wp-block-image size-large"><img src="http://localhost:8889/wp-content/uploads/2024/06/car-1024x576.jpeg" alt="" class="wp-image-20"/></figure>
<!-- /wp:image -->
```

In [this commit](https://github.com/WordPress/performance/pull/1252/commits/2656a0d355e8683ddb047cf8f5c5f06d1311aed0), I added support for image resizing (resizing allowed only in the default alignment).